kvbc/src/kvbc_app_filter/kvbc_app_filter.cpp

```diff
@@ -355,236 +354,197 @@ uint64_t KvbAppFilter::newestTagSpecificPublicEventGroupId() const {
   return public_newest + private_newest;
 }
 
-std::optional<uint64_t> KvbAppFilter::getNextEventGroupId(std::shared_ptr<EventGroupClientState> &eg_state) {
-  LOG_DEBUG(logger_, "Public_offset: " << eg_state->public_offset << ", private_offset: " << eg_state->private_offset);
-  LOG_DEBUG(logger_, "Event group id list batch size: " << eg_state->event_group_id_batch.size());
+FindGlobalEgIdResult KvbAppFilter::findGlobalEventGroupId(uint64_t external_event_group_id) const {
+  auto external_oldest = oldestTagSpecificPublicEventGroupId();
+  auto external_newest = newestTagSpecificPublicEventGroupId();
+  ConcordAssertNE(external_oldest, 0);  // Everything pruned or was never created
```

Comment:
Since `oldestTagSpecificPublicEventGroupId()` can return 0, maybe we should be explicit and put a precondition on `findGlobalEventGroupId()` so that it is clear when the use can call it.

Reply:
I will add a comment to the function. Thanks!